node/pkg/proto/gossip/v1/gossip.pb_test.go

```diff
@@ -0,0 +1,119 @@
+package gossipv1
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/hex"
+	vaa "github.com/certusone/wormhole/node/pkg/vaa"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/assert"
+	"testing"
+
+	"time"
+)
+
+func getVAA() vaa.VAA {
+	var payload = []byte{97, 97, 97, 97, 97, 97}
+	var governanceEmitter = vaa.Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+
+	vaa := vaa.VAA{
+		Version:          uint8(1),
+		GuardianSetIndex: uint32(1),
+		Signatures:       nil,
+		Timestamp:        time.Unix(0, 0),
+		Nonce:            uint32(1),
+		Sequence:         uint64(1),
+		ConsistencyLevel: uint8(32),
+		EmitterChain:     vaa.ChainIDSolana,
+		EmitterAddress:   governanceEmitter,
+		Payload:          payload,
+	}
+
+	return vaa
+}
+
+func TestSignedObservation(t *testing.T) {
+	vaa := getVAA()
+	digest := vaa.SigningMsg()
+
+	txhash := []byte{97, 97, 97, 97, 97, 97}
+	signature := []byte{97, 97, 97, 97, 97, 97}
+
+	privKey, _ := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
+
+	obsv := SignedObservation{
+		Addr:      crypto.PubkeyToAddress(privKey.PublicKey).Bytes(),
+		Hash:      digest.Bytes(),
+		Signature: signature,
+		TxHash:    txhash,
+		MessageId: vaa.MessageID(),
+	}
+
+	assert.Equal(t, signature, obsv.Signature)
+	assert.Equal(t, txhash, obsv.TxHash)
+	assert.Equal(t, vaa.MessageID(), obsv.MessageId)
+}
+
+func getSignedObservation() SignedObservation {
+	vaa := getVAA()
+	digest := vaa.SigningMsg()
+
+	txhash := []byte{97, 97, 97, 97, 97, 97}
+	signature := []byte{97, 97, 97, 97, 97, 97}
+
+	privKey, _ := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
+
+	return SignedObservation{
+		Addr:      crypto.PubkeyToAddress(privKey.PublicKey).Bytes(),
+		Hash:      digest.Bytes(),
+		Signature: signature,
+		TxHash:    txhash,
+		MessageId: vaa.MessageID(),
+	}
+}
+
+func TestSignedObservation_String(t *testing.T) {
+	obsv := getSignedObservation()
+	obsvString := obsv.String()
+	assert.Contains(t, obsvString, "addr")
+	assert.Contains(t, obsvString, "hash")
+	assert.Contains(t, obsvString, "signature")
+	assert.Contains(t, obsvString, "tx_hash")
+	assert.Contains(t, obsvString, "message_id")
+	assert.Greater(t, len(obsvString), 200)
+	assert.Less(t, len(obsvString), 300)
+}
```

Comment:
It sucks that you're generating a new private key every time in `getSignedObservation()`. If it was quasi-static, you could simply assert the string contents, which would be a better test overall.

This is solid given the value changes eachtime, but could miss things in the future (unlikely), but less magic and more bare asserts are good, as @cjw296 taught me.